docs/articles/rpc.rst

```diff
@@ -0,0 +1,237 @@
+                    'demo-rpc-server' => [
+                        'callback' => 'timestwo',
+                        'queue' => 'demo-rpc-server',
+                        'auto_setup_fabric' => true,
+                    ],
+                ],
+            ],
+        ],
+    ];
+
+So what's important here? The JSON RPC Server needs an exchange and a queue. All messages routed to the exchange, will
+get routed to the server queue. In this example we use a direct exchange and a single queue for the server. This is the
+pretty much simplest setup we can have.
+
+The second this is, the server needs a callback, we use `timestwo` here. As you can see in the dependencies setup, the
+callback is simply turning a Request into a Response like this:
+
+.. code-block:: php
+
+    function (\Humus\Amqp\JsonRpc\Request $request): \Humus\Amqp\JsonRpc\Response {
+        return \Humus\Amqp\JsonRpc\JsonRpcResponse::withResult($request->id(), $request->params() * 2);
+    }
+
+For the callback function consider this:
+
+- Return an instance of `Humus\Amqp\JsonRpc\Response`
+- Just return some value, the server will automatically wrap the result with an instance of `Humus\Amqp\JsonRpc\JsonRpcResponse`
+- All thrown exceptions will return an error response to the client
+
+The `Humus\Amqp\JsonRpc\Request` also has a method named `method()` - this allows you to a single callback to return
+different results, based on the method. For example:
+
+.. code-block:: php
+
+    function (\Humus\Amqp\JsonRpc\Request $request): \Humus\Amqp\JsonRpc\Response {
+        switch ($request->method()) {
+            case 'times2':
+                return \Humus\Amqp\JsonRpc\JsonRpcResponse::withResult($request->id(), $request->params() * 2);
+            case 'times3:
+                return \Humus\Amqp\JsonRpc\JsonRpcResponse::withResult($request->id(), $request->params() * 3);
+            case 'plus5':
+                return \Humus\Amqp\JsonRpc\JsonRpcResponse::withResult($request->id(), $request->params() + 5);
+            default:
+                return \Humus\Amqp\JsonRpc\JsonRpcResponse::withError($request->id(), new \Humus\Amqp\JsonRpc\JsonRpcError(32601));
+        }
+    }
+
+Running JSON-RPC servers
+------------------------
+
+To start a JSON-RPC server
+
+.. code-block:: bash
+
+    $ ./vendor/bin/humus-amqp json_rpc_server -n demo-rpc-server -a 100
+
+This will start the `demo-rpc-server` and consume 100 messages until if stops or times out.
+
+Set up JSON RPC Client
+----------------------
+
+Again, let's start with a sample configuration first (and skip the server config part, to make it easier to read):
+
+.. code-block:: php
+
+    return [
+        'dependencies' => [
+            'factories' => [
+                Driver::class => Container\DriverFactory::class,
+                'default-amqp-connection' => [Container\ConnectionFactory::class, 'default'],
+                'demo-rpc-client' => [Container\JsonRpcClientFactory::class, 'demo-rpc-client'],
```

Comment:
Do you not find that a bit of throwing? I'd understand `[Container\JsonRpcClientFactory::class, 'demoRpcClient']` because it looks more like a method name.